make.js

```diff
@@ -98,6 +99,37 @@ target.clean = function () {
     rm('-Rf', path.join(__dirname, '_test'));
 };
 
+//
+// Generate documentation (currently only YAML snippets)
+// ex: node make.js gendocs
+// ex: node make.js gendocs --task ShellScript
+//
+target.gendocs = function() {
+
+    var docsDir = path.join(__dirname, '_gendocs');
+    rm('-Rf', docsDir);
+    mkdir('-p', docsDir);
+    console.log();
+    console.log('> generating docs');
+
+    taskList.forEach(function(taskName) {
+        var taskPath = path.join(__dirname, 'Tasks', taskName);
+        ensureExists(taskPath);
+
+        // load the task.json
+        var taskJsonPath = path.join(taskPath, 'task.json');
+        if (test('-f', taskJsonPath)) {
+            var taskDef = require(taskJsonPath);
+            validateTask(taskDef);
+
+            // create YAML snippet
+            createYamlSnippet(taskDef, path.join(docsDir, taskName + '.yml'));
```

Comment:
taskName is the folder name? is that intentional?

or would it make more sense to use the name and major version from within the task.json?

Reply:
YAML snippets are now output as files in this path pattern that includes the major version:
`_gendocs/ArchiveFiles.2.yml`